sympy/polys/series/tests/test_powerseriesring.py

```diff
@@ -0,0 +1,486 @@
+
+
+def test_int_inversion(rd_int):
+    R, x, one, R3, x3, one3, R10, x10, one10, _ = rd_int
+
+    assert same(R, R.inversion(R.add(one, x)), [1, -1, 1, -1, 1, -1], 6)
+    assert same(
+        R,
+        R.inversion(R.add(R.pow_int(R.multiply(R.add(one, x), x), 3), one)),
+        [1, 0, 0, -1, -3, -3],
+        6,
+    )
+    assert same(R3, R3.inversion(R3.from_list([1, 3, -2])), [1, -3, 11], 3)
+    assert same(
+        R10,
+        R10.inversion(R10.from_list([1, -2, 3])),
+        [1, 2, 1, -4, -11, -10, 13, 56, 73, -22],
+        10,
+    )
+
+    raises(NotReversible, lambda: R.inversion(R.zero))
+    raises(NotReversible, lambda: R.inversion(R.from_list([0, 1, 2])))
+
+
+def test_rational_inversion(rd_rational):
+    R, _, _, R3, _, _, R10, *_ = rd_rational
+
+    f_r = R.from_list([QQ(2, 3), QQ(-1, 4), QQ(3, 5)])
+    assert same(
+        R,
+        R.inversion(f_r),
+        [
+            (3, 2),
+            (9, 16),
+            (-729, 640),
+            (-4779, 5120),
+            (138267, 204800),
+            (1791153, 1638400),
+        ],
+        6,
+    )
+
+    f3 = R3.from_list([QQ(1, 2), QQ(-3, 4), QQ(5, 6)])
+    assert same(R3, R3.inversion(f3), [(2, 1), (3, 1), (7, 6)], 3)
+
+    f10 = R10.from_list([QQ(3, 5), QQ(1, 7), QQ(-2, 9)])
+    assert same(
+        R10,
+        R10.inversion(f10),
+        [
+            (5, 3),
+            (-25, 63),
+            (2825, 3969),
+            (-26375, 83349),
+            (1779875, 5250987),
+            (-7274375, 36756909),
+            (1199485625, 6947055801),
+            (-16690759375, 145888171821),
+            (838109346875, 9190954824723),
+            (-12369018828125, 193010051319183),
+        ],
+        10,
+    )
+
+
+def test_int_reversion(rd_int):
+    R, x, one, R3, x3, one3, R10, x10, one10, _ = rd_int
+
+    assert same(R, R.reversion(x), [0, 1], 6)
+    assert same(R, R.reversion(R.multiply(R.add(one, x), x)), [0, 1, -1, 2, -5, 14], 6)
+    assert same(
+        R,
+        R.reversion(R.from_list([0, 1, 53, 2, 1, 3, 2])),
+        [0, 1, -53, 5616, -743856, 110349083],
+        6,
+    )
+    assert same(R3, R3.reversion(R3.multiply(R3.add(one3, x3), x3)), [0, 1, -1], 3)
+    assert same(
+        R10,
+        R10.reversion(R10.from_list([0, 1, 2, -1, 3])),
+        [0, 1, -2, 9, -53, 347, -2429, 17808, -134991, 1049422],
+        10,
+    )
+
+    raises(NotReversible, lambda: R.reversion(R.zero))
+    raises(NotReversible, lambda: R.reversion(R.from_list([0, 0, 2])))
+
+
+def test_rational_reversion(rd_rational):
+    R, _, _, R3, _, _, R10, *_ = rd_rational
+
+    f1 = R.from_list([QQ(0, 1), QQ(3, 2), QQ(1, 4), QQ(-2, 5)])
+    assert same(
+        R,
+        R.reversion(f1),
+        [(0, 1), (2, 3), (-2, 27), (116, 1215), (-106, 2187), (24604, 492075)],
+        6,
+    )
+
+    f3 = R3.from_list([QQ(0, 1), QQ(5, 4), QQ(-1, 3)])
+    assert same(R3, R3.reversion(f3), [(0, 1), (4, 5), (64, 375)], 3)
```

Comment:
I think we probably want a convenience function `R.__call__` to do:
```
f3 = R3([(0,1), (5,4), (-1,3)])
```
rather than
```
f3 = R3.from_list([QQ(0, 1), QQ(5, 4), QQ(-1, 3)])
```